### ISSUE TITLE
Add my own model predition?

### ISSUE THREAD

**Issue by raulocho** [justadudewhohacks/face-api.js#691]
Hi there, first of al congratulations and thank you for the great project that you share with all us.

Im kinda new in the area,  so sorry about something i can wrong.

I had trained a facemaks model to detect if some face have a mask, not have or is bad wearing it.

I would like to add / concat the result to faceapi analys im trying figure how do it,  anyone can give a hand with that? would be great have al the results togheter.

Thank you in advance.

**Comment by JediahDizon** [comment]
What you need is called "custom classifier" and there is a tutorial here:

https://codelabs.developers.google.com/codelabs/tensorflowjs-teachablemachine-codelab/index.html#0

**Comment by raulocho** [comment]
@JediahDizon Thank You, I have already the model trained, just want create a new function like ".detectFacemask()" that can be concatened to the rest of faceapi functions and add the result to the json return.

**Comment by JediahDizon** [comment]
Ahh I see. Well I'm not sure how the face-api do its black box magic so I don't know about diving deep in the face-api code to add a new function to detect a mask.

However, if I were to implement it on the top of my head, I would make it so that I can have the classifier run independently with the face-api library. 

So you got two sensors for a face, one that detects the face, and one that identifies if a mask exist. Benefit of this is when the mask is not a human face and cannot be detected by face-api, you would still know there is a mask there.

Downside is a hit in performance because you have two neural networks running at the same time. I dunno how bad, it could be insignificant or drastic. So I would  test it out if this solution is plausible.

**Comment by raulocho** [comment]
Well finally i found a way to do it, just adding some code and a function.

Thx for the help anyway.

**Comment by JediahDizon** [comment]
That's some great work. You should make a PR so that the devs can add this new features to the library. I'm curious about the implementation too.

**Comment by Reaper622** [comment]
@raulocho  I'm doing a system to detect a mask now, see that you already have a solution. I wonder if this part of the code is open source, where can I find it?
